tests/test_utility.py

```diff
@@ -17,17 +17,24 @@
 
 
 def test_show_version():
-    with pytest_warns_bounded(
-        UserWarning,
-        match="Setuptools is replacing distutils",
-        upper="3.11.99",
-        version_str=platform.python_version(),
-    ):
-        context: AbstractContextManager
-        if PD_LTE_22 and NUMPY20:  # https://github.com/PyTables/PyTables/issues/1172
+    """Test show_versions method types with split case for pandas and python versions."""
+    context: AbstractContextManager = nullcontext()
+    if PD_LTE_22:
+        # distutils warning is only raised with pandas<3.0.0
+        if NUMPY20:  # https://github.com/PyTables/PyTables/issues/1172
             context = pytest.raises(ValueError)
-        else:
-            context = nullcontext()
+        with (
+            pytest_warns_bounded(
+                UserWarning,
+                match="Setuptools is replacing distutils",
+                upper="3.11.99",
+                version_str=platform.python_version(),
+            ),
+            context,
+        ):
+            check(assert_type(pd.show_versions(True), None), type(None))
+            check(assert_type(pd.show_versions(False), None), type(None))
+    else:
         with context:
```

Comment:
Since the `else` will get entered with `context = nullcontext()`, you don't need to do the test with a context manager.  So you can delete this line.

Reply:
Correct that was unnecessary, let me remove!